Code/Main/shared/src/main/scala/rescala/extra/simpleprop/SimpleScheduler.scala

```diff
@@ -12,12 +13,14 @@ trait SimpleStruct extends Struct {
 
```

Comment:
Refactor the changes to SimpleState and SimpleScheduler into subclasses, so that the original scheduler code is preserved.

Reply:
> As far as I can see, the changes to the macro code are removed? or did I just miss them?

We removed the `TestSpecification` macro. Should we add it back in?

> Refactor the changes to SimpleState and SimpleScheduler into subclasses, so that the original scheduler code is preserved.

I tried this refactoring for the last few hours but could not manage it. The first problem is that `SimpleScheduler` is an object and can therefore not be subclassed. I converted it into a trait that was then used by both the original scheduler as well as our custom one. However, our only real change to the scheduler object is the added line `Util.evaluateInvariants(created ++ sorted, initialWrites)` in `forceNewTransaction` (see https://github.com/rescala-lang/REScala/pull/32/commits/932f37b76f8d6a8dda994a504031199926b08ac2#diff-bd20bad8133f5826539b1a478a543c2dR127). This would mean that we either had to override the whole method again (which is basically the whole scheduler) or change to base implementation to add a hook-like function at this line that could then be overridden by our scheduler. I'd prefer the later approach and call the hook something like `onEvaluationComplete`.

The bigger problem is subclassing the state. I was not able to change the `outgoing` and `incoming` fields in our subclassed `SimpleState` to contain our custom `SimpleStruct` without changing the signature so drastically that a whole lot of other problems appeared.

@rmgk maybe you know a way to do this in a much more elegant way. For now I simply reverted our changes to the original `SimpleScheduler` and moved our modified version to `rescala.extra.simpleprop.invariant` (see https://github.com/rescala-lang/REScala/pull/32/commits/932f37b76f8d6a8dda994a504031199926b08ac2#diff-bd20bad8133f5826539b1a478a543c2d).